inference-engine/src/multi_device/multi_device_exec_network.cpp

```diff
@@ -158,106 +158,183 @@ MultiDeviceExecutableNetwork::MultiDeviceExecutableNetwork(const std::string&
+    if (isActualDevCPU) {
+         _loadContext[CPU].isEnabled = false;
+    } else {
+        const auto CPUIter = std::find_if(metaDevices.begin(), metaDevices.end(),
+                [=](const DeviceInformation& d)->bool{return d.deviceName.find("CPU") != std::string::npos;});
+        // if have CPU Device,  enable _loadContext[CPU]
+        if (CPUIter != metaDevices.end()) {
+            _loadContext[CPU].isEnabled = true;
+            _loadContext[CPU].deviceInfo = *CPUIter;
+        } else {
+            _loadContext[CPU].isEnabled = false;
+        }
+    }
+
+    // initialize the rest members of load context
+    for (int i = 0; i < CONTEXTNUM; i++) {
+         if (_loadContext[i].isEnabled) {
+             _loadContext[i].future =  _loadContext[i].promise.get_future();
+              auto* contextPtr = &_loadContext[i];
+             _loadContext[i].task = [this, contextPtr, modelPath, network]() mutable {
+                      TryToLoadNetWork(*contextPtr, modelPath, network);
+                      if (contextPtr->isLoadSuccess) {
+                          GenerateWorkers(contextPtr->deviceInfo.deviceName, contextPtr->executableNetwork);
+                          //need lock
+                          {
+                             std::lock_guard<std::mutex> lock(_confMutex);
+                             _config.insert(contextPtr->deviceInfo.config.begin(),
+                                            contextPtr->deviceInfo.config.end());
+                          }
+                          contextPtr->isAlready = true;
+                      }
+                      contextPtr->promise.set_value();
+                      // the first load network process finished
+                      std::call_once(_firstLoadOC, [this] () {
+                              _firstLoadPromise.set_value();
+                              });
+             };
+         }
+    }
+
+    if (_loadContext[CPU].isEnabled) {
+        _firstLoadFuture = _firstLoadPromise.get_future();
         // will not wait for loading accelerator network,
-        // so some parameters need to be transferred by value.
-       _executor->run([&, modelPath, network, device, deviceConfig]() {
-            SoExecutableNetworkInternal executableNetwork;
-            if (!modelPath.empty()) {
-                executableNetwork = _core->LoadNetwork(modelPath, device, deviceConfig);
-            } else {
-                executableNetwork = _core->LoadNetwork(network, device, deviceConfig);
-            }
+        // so the executor can't be destroyed before finished the task,
+        // so use executor as a member of MultiDeviceExecutableNetwork.
+        _executor = InferenceEngine::ExecutorManager::getInstance()->getIdleCPUStreamsExecutor(
+                IStreamsExecutor::Config{"AutoDeviceAsyncLoad",
+                static_cast<int>(std::thread::hardware_concurrency()) /* max possible #streams*/,
```

Comment:
I don't know whether it would affect the performance if set to 2, we can test it after our performance tests are setup totally, Currently just not modify it  from my point of view.